File: src/data/CommunicationData.cpp
//
// Created by Andrei on 16.03.2021.
//

#include <comm/data/CommunicationData.h>

using namespace comm;

CommunicationData::CommunicationData() : serializeState(0), deserializeState(0) {}

CommunicationData::~CommunicationData() = default;

void CommunicationData::resetSerializeState() {
    this->serializeState = 0;
}

char *CommunicationData::getDeserializeBuffer() {
    return nullptr;
}

void CommunicationData::resetDeserializeState() {
    this->deserializeState = 0;
}

File: csharp/data/CommunicationData.cs
namespace DesignAIRobotics.Comm.data {
    public abstract class CommunicationData {
        public CommunicationData() {
            this.serializeState = 0;
            this.deserializeState = 0;
        }

        ~CommunicationData() { }

        public abstract MessageType getMessageType();

        public void resetSerializeState() {
            this.serializeState = 0;
        }

        public abstract bool serialize(utils.Buffer buffer, int start, bool forceCopy, bool verbose);

        public abstract ulong getExpectedDataSize();

        public virtual byte[] getDeserializeBuffer() {
            return null;
        }

        public void resetDeserializeState() {
            this.deserializeState = 0;
        }

        public abstract bool deserialize(utils.Buffer buffer, int start, bool forceCopy, bool verbose);

        protected int serializeState, deserializeState;
    };
}
